uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/payment/adyen/AdyenPaymentPresenter.kt

```diff
@@ -80,18 +80,31 @@ class AdyenPaymentPresenter(view: PaymentDropInContract.Actions,
         if (resultCode == AppCompatActivity.RESULT_OK && data == null) {
             view?.showPaymentFailureDialog()
         } else if (resultCode == AppCompatActivity.RESULT_OK && data != null) {
-            val dataString = data.getStringExtra(AdyenResultActivity.RESULT_KEY) ?: ""
-            val payload = JSONObject(dataString)
-            when (payload.optString(RESULT_CODE, "")) {
-                AdyenPaymentView.AUTHORISED -> {
-                    this.tripId = payload.optString(TRIP_ID, "")
-                    updateCardDetails(paymentData = payload.optString(ADDITIONAL_DATA, ""))
-                }
-                else -> {
-                    val error = convertToKarhooError(payload)
-                    view?.showPaymentFailureDialog(error)
+
+            //todo error reason map out to KHError
```

Comment:
There are several TODOs throughout this PR. Shouldn't we take care of them at this stage?

Reply:
Not sure about the others, this one will be removed since it's handled below at lines 102-105